### ISSUE TITLE
Tweet attrition testing

### ISSUE THREAD

**Issue by martysteer** [medialab/minet#375]
Hi @Yomguithereal,

I've done a bit of testing on the attrition command and it's really great. Found a few things worth mentioning, but otherwise it works really well, and is comparable to the twarc1 script. Since we last corresponded, my twarc1 credentials changed, and I ended up finding out that twarc2 (and Twitter!) have implemented a 'compliance' mechanism for batching this stuff. Geez... anyway, I tested the [twarc2 compliance-job command](https://twarc-project.readthedocs.io/en/latest/twarc2_en_us/#compliance-job) and while that works, using the results is a bit trickier because it splits users and tweets into separate requests. The minet attrition command combines these perfectly, and for my project's data structures, it's much much easier to strip out tweet and user ids from my csv files for processing with `minet attrition`.

Steps:
Fresh installed using scripts/install.sh. 

```
$ minet --version        
minet 0.55.9
```

My test CSV contains two columns, 'a,d', tweet id and user id respectively. I tried the example commands in the docs and got an error:

```
$ minet twitter attrition a corpus-attrition.csv --user d --ids > minet-attrition-report.csv 
usage: minet [-h] [--version] [--rcfile RCFILE]
             {buzzsumo,bz,cookies,crawl,crowdtangle,ct,extract,facebook,fb,fetch,google,hyphe,mediacloud,mc,resolve,scrape,twitter,tw,url-extract,url-join,url-parse,youtube,yt,help}
             ...
minet: error: unrecognized arguments: --user d
```

Checking the command line help it says there are three positional arguments, tweet_column, user_column and file. If I follow this, it works. So the latest online docs are a bit out of sync with the command in the master branch (but I can see a bunch of work going on behind the scenes so presume this is still in progress).

This command worked beautifully:

```
$ minet twitter attrition a d corpus-attrition.csv --ids -o minet-attrition-report.csv --resume
```

however the resuming didn't currently work. It doesn't show an error, but just duplicates the results in the same output file, which is not a big deal but thought this was worth mentioning.

Here's the counts from the output file. Took 8 minutes to generate the report:

```
$ cut -d, -f3 minet-attrition-report.csv | sort | uniq -c
6705 available_tweet
 237 deactivated_user
  73 protected_user
 631 suspended_user
   1 tweet_current_status
 210 unavailable_tweet
```

As for mapping to the previous data I collected using twarc1, here is the category mapping I used, incase it helps others. The individual totals differed, but the grand total was identical.

[twarc1](https://github.com/DocNow/twarc/blob/v1.12.1/utils/deletes.py) | [minet attrition](https://github.com/medialab/minet/blob/master/docs/cli.md#attrition)
-- | --
.  | available_tweet
TWEET_DELETED | unavailable_tweet
RETWEET_DELETED|"
ORIGINAL_TWEET_DELETED|"
USER_SUSPENDED | suspended_user
ORIGINAL_USER_SUSPENDED|"
USER_PROTECTED | protected_user
ORIGINAL_USER_PROTECTED|"
USER_DELETED | deactivated_user
ORIGINAL_USER_DELETED|"

Thanks so much specifically to @ameliepelle for all your work on this. it's super useful (and obviously important, because twitter has been working on it as well - [their docs are here](https://developer.twitter.com/en/docs/twitter-api/compliance/batch-compliance/introduction)).

**Comment by Yomguithereal** [comment]
Hello @martysteer,

regarding the missing `--user` flag, it was refactored thusly some days ago by @ameliepelle and I did not release it yet, so there is indeed a discrepancy between the docs on the repo and the latest `0.55.9` version, sorry. I will release `0.56.0` asap to avoid this problem. @ameliepelle also added a way to work using tweet urls directly as a convenience.

Regarding the `--resume` issue I don't remember we implemented it on this command. Let me check what went wrong.

I am happy to know it seems to work as well as expected in any case :)

**Comment by Yomguithereal** [comment]
@martysteer the v0.56 is building right now and should be available in some minutes.

Regarding the resuming, we did not succeed in reproducing your issue, as it works on both our end with @ameliepelle. Can you give us a file demonstrating the problem?

Also, we were wondering if you saw the `--retweeted-id` flag, that might enhance the quality of your results wrt retweets if you have the necessary data.

**Comment by martysteer** [comment]
Nice. Will take a look at the next version tomorrow and try to address your points.

Yep, saw the retweeted id flag, but don't have my data in easy format for that yet. The tweet URL's is a nice option.

Cheers!

**Comment by martysteer** [comment]
I've updated to v0.56 and tested again and the documented syntax works well. Thankyou!

Caveat with my testing: I only tested with tweet and user id's, not with URL's or retweet id's.

As for the `--resume` flag, not sure what I did the other day but I can't reproduce either, so happy to say this is working for me in v0.56.

I plan to time how long it takes to run across my corpus of 12m tweets because minet's implementation seems much faster and has categories which are directly comparable. (twarc1 took several months. Well, it wasn't twarc1's fault, it was because of all the 15 minute waits my account was subjected to. ;-) ).
